src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -914,19 +914,29 @@ public async Task SslStream_ClientCertificate_SendsChain()
             }
         }
 
-        [Fact]
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/68206", TestPlatforms.Android)]
-        public async Task SslStream_ClientCertificateContext_SendsChain()
+        public async Task SslStream_ClientCertificateContext_SendsChain(bool useTrust)
         {
             (X509Certificate2 clientCertificate, X509Certificate2Collection clientChain) = Configuration.Certificates.GenerateCertificates(nameof(SslStream_ClientCertificateContext_SendsChain), serverCertificate: false);
             TestHelper.CleanupCertificates(nameof(SslStream_ClientCertificateContext_SendsChain));
 
+            SslCertificateTrust? trust = null;
+            if (useTrust)
+            {
+                // This is simplification. We make all the intermediates trusted,
+                // normally just the root would go here.
+                trust = SslCertificateTrust.CreateForX509Collection(clientChain, false);
```

Comment:
The root is part of the `clientChain` collection, right? So it should be enough to just take the last/first item in the collection to remove the simplification. Or am I missing something else?

Reply:
The problem is that the collection is just bunch of certificates and it is not neeccesatily ordered (like the X509Chain.ChainElements). It can also contain more or less  certificates and/or unrelated certificates. It is job of `X509Chain.Build` to sort that out and construct specific certificate array.